lib/telephony/otp_sender.rb

```diff
@@ -1,14 +1,16 @@
 module Telephony
   class OtpSender
-    attr_reader :recipient_phone, :otp, :expiration, :channel, :domain, :country_code
+    attr_reader :recipient_phone, :otp, :expiration, :channel, :domain, :country_code,
+                :extra_metadata
 
-    def initialize(to:, otp:, expiration:, channel:, domain:, country_code:)
+    def initialize(to:, otp:, expiration:, channel:, domain:, country_code:, extra_metadata:)
       @recipient_phone = to
       @otp = otp
       @expiration = expiration
       @channel = channel.to_sym
       @domain = domain
       @country_code = country_code
+      @extra_metadata = extra_metadata
```

Comment:
could/should we parse these from `to:`? (basically everything except `resend`)

```suggestion
      parsed_phone = Phonelib.parse(to)
      @extra_metadata = {
        area_code: parsed_phone.area_code,
        phone_fingerprint: Pii::Fingerprinter.fingerprint(parsed_phone.e164),
      }
```

Reply:
We could yeah, but it's already being parse (not sure how expensive it is)

Reply:
it also includes some information that isn't derived from the phone number itself